src/Evolution/Systems/RadiationTransport/M1Grey/Sources.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// \cond
+class DataVector;
+
+namespace RadiationTransport {
+namespace M1Grey {
+
+/*!
+ * \brief Compute the curvature source terms for the flux-conservative
+ * grey M1 radiation transport.
+ *
+ *
+ * A flux-conservative system has the generic form:
+ * \f[
+ * \partial_t U_i + \partial_m F^m(U_i) = S(U_i)
+ * \f]
+ *
+ * where \f$F^a()\f$ denotes the flux of a conserved variable \f$U_i\f$ and
+ * \f$S()\f$ denotes the source term for the conserved variable.
+ *
+ * For the grey M1 formalism (neglecting coupling to the fluid):
+ * \f{align*}
+ * S({\tilde E}) = & \alpha \tilde P^{ij} K_{ij} - \tilde F^i \partial_i \alpha,
+ * S({\tilde F_i}) = & -\tilde E \partial_i \alpha + \tilde F_k \partial_i
+ * \beta^k
+ * + \frac{1}{2} \lapse \tilde P^{ij} \partial_i \gamma_{jk},
+ * \f}
+ *
+ * where \f${\tilde E}\f$, \f${\tilde F_i}\f$, \f${\tilde P}^{ij}\f$ are the
+ * densitized energy, momentum, and pressure tensor of the neutrinos/photons,
+ * \f$K^{ij}\f$ is the extrinsic curvature, and \f$\alpha\f$, \f$\beta^i\f$,
+ * \f$\gamma^{ij}\f$ are the lapse, shift and 3-metric.
+ */
+static void apply(
```

Comment:
Please make this a struct over all neutrino species, with the following apply function that calls your function which should be in a `detail` namespace and renamed `compute_sources_impl`
```cpp
template <typename... NeutrinoSpecies>
struct ComputeSources {
  static void apply(const gsl::not_null<db::item_type<NeutrinoSpecies>*>... sources_tilde_e,
    const gsl::not_null<db::item_type<TildeF<NeutrinoSpecies>>*>... sources_tilde_f,
    const db::item_type<NeutrinoSpecies>&... tilde_e,
   /* the remaining args */) noexcept {
       EXPAND_PACK_LEFT_TO_RIGHT(
          detail::compute_sources_impl(sources_tilde_e, sources_tilde_f, tilde_e, 
             /*the remaining args*/));
   }
};
```

Reply:
After the "EXPAND_PACK_LEFT_TO_RIGHT" command, what's the structure of, e.g., sources_tilde_e? i.e. how do I access each species after that command?